frontend/packages/ux-editor/src/components/config/editModal/EditDataModelBindings.tsx

```diff
@@ -43,28 +47,117 @@ export const EditDataModelBindings = ({
   };
 
   const { uniqueKey, key, label } = renderOptions || {};
+  const [linkIconVisible, setLinkIconVisible] = useState(true);
+  const [dropdownVisible, setDropdownVisible] = useState(false);
+  const [CheckmarkAndTrashIconsVisible, setCheckmarkAndTrashIconsVisible] = useState(false);
+  const [checkmarkClicked, setCheckmarkClicked] = useState(false);
+  const [isLinkedDatamodelHovered, setIsLinkedDatamodelHovered] = useState(false);
+  const handleHoverActive = () => setIsLinkedDatamodelHovered(true);
+  const handleHoverDisabled = () => setIsLinkedDatamodelHovered(false);
```

Comment:
Here are a lot of useStates which can become hard to maintain and understand which state should be toggle to true/false based on different situations.  I have some thoughts about how we can refactor this to be a little simpler. 

I would share my thoughts tomorrow. It's time for bed now. 😴

Reply:
I have been thinking about this all night, hehe 😂 I believe that we need this feature for many input fields around in Studio. I think we should separate this logic within a reusable component, maybe we can name it `<InputActionWrapper/>`. 

I have created a Sandbox that shows how we can implement such component, https://codesandbox.io/s/react-typescript-forked-qdkmsw?file=/src/InputActionWrapper.tsx. 😊 


Few key-things I have done:

1. Reduced from 5 useStates to just one. 
2. Created ActionGroups that has logical name based on which actions should be available within different situations
3. The `<InputActionWrapper/>` is re-usable.

Eksample of usage:

```jsx
 <InputWrapperAction
        onDeleteClick={() => console.log("delete action triggerd")}
        onSaveClick={() => console.log("save action triggered")}
      >
        <input type="text" />
  </InputWrapperAction>
```